### PR TITLE
Add recipe for EndlessSky

### PR DESCRIPTION
**Pull request by Vrondir**
Recipe builds fine using secondary architectures, and the game launches, however the game itself crashes on a nightly but runs fine on r1Beta1, don't really know why.

### CODE REVIEW

**Comment by Vrondir** [comment]
(Update) There was a nasty glitch with the cursor so I added a patchset for that :)

**Comment by scottmc** [comment]
The nightly vs R1Beta1 might be due to one or more of the dependencies libraries.  You might check other recipes, as there's been updates to libpng and the SDL libs, perhaps others that might require recipe adjustments.
Can you post a screenshot of the game running in Haiku?

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libglew$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libmad$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
```
Comment (Begasus) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]: Suggestion:
```
	lib:libGL$secondaryArchSuffix
	lib:libglew$secondaryArchSuffix
	lib:libjpeg$secondaryArchSuffix
	lib:libmad$secondaryArchSuffix
	lib:libopenal$secondaryArchSuffix
	lib:libpng16$secondaryArchSuffix
	lib:libSDL2_2.0$secondaryArchSuffix
```

Comment (Begasus) [reply games-action/endless_sky/endless_sky-0.9.10.recipe]: Same for devel:*

Comment (Vrondir) [reply games-action/endless_sky/endless_sky-0.9.10.recipe]: @Begasus that still doesn't work on Nightly.

@scottmc here's the game crashing on startup in a Nightly 32bit:https://snipboard.io/4O2RqZ.jpg and what it outputs in the debugger:  https://snipboard.io/NjYOwJ.jpg
and here is it running on r1Beta1 32bit: https://snipboard.io/ZfeSma.jpg

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+COPYRIGHT="Michael Zahniser"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/endless-sky/endless-sky/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="58f2e405e171be8426004216df4df7bb3a2792a507a5d01f77a0f02346332f6f"
+SOURCE_DIR="endless-sky-$portVersion"
+PATCHES="endless_sky-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 !x86_64"
+SECONDARY_ARCHITECTURES="x86"
```
Comment (Begasus) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]: Mark architecture as "?" so the buildbot doesn't try to build it

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+PROVIDES="
+	endless_sky$secondaryArchSuffix = $portVersion
+	app:EndlessSky$secondaryArchSuffix = $portVersion
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	app:EndlessSky = $portVersion
```

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	lib:libmad$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+"
+
+BUILD_REQUIRES="
+	devel:libglew$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	devel:libjpeg$secondaryArchSuffix >= 9
```

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	devel:libopenal$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+"
+
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:gcc$secondaryArchSuffix
+	cmd:scons
+"
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	"
```

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+BUILD_REQUIRES="
+	devel:libglew$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libmad$secondaryArchSuffix
+	devel:libopenal$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+"
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	"
```

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libglew$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libmad$secondaryArchSuffix
+	lib:libopenal$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+"
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	"
```

File: games-action/endless_sky/endless_sky-0.9.10.recipe [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+SOURCE_DIR="endless-sky-$portVersion"
+PATCHES="endless_sky-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 !x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	endless_sky$secondaryArchSuffix = $portVersion
+	app:EndlessSky$secondaryArchSuffix = $portVersion
+"
```
Comment (korli) [inline games-action/endless_sky/endless_sky-0.9.10.recipe]:
```suggestion
	"
```

File: games-action/endless_sky/patches/endless_sky-0.9.10.patchset [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/source/main.cpp b/source/main.cpp
+index 323b594..1516d3c 100644
+--- a/source/main.cpp
++++ b/source/main.cpp
+@@ -329,7 +329,7 @@ int main(int argc, char *argv[])
+ 			if(shouldShowCursor != showCursor)
+ 			{
+ 				showCursor = shouldShowCursor;
+-				SDL_ShowCursor(showCursor);
++				//SDL_ShowCursor(showCursor);
```
Comment (korli) [inline games-action/endless_sky/patches/endless_sky-0.9.10.patchset]: Better with 
#ifndef \_\_HAIKU\_\_

File: games-action/endless_sky/endless_sky-0.9.10.recipe (hunk 1 above)
Comment (korli) [reply games-action/endless_sky/endless_sky-0.9.10.recipe]: It crashes also on R1Beta1 64bits

**Pull request merged**